src/Components/Popover.tsx

```diff
@@ -0,0 +1,161 @@
+import React from "react"
+import ReactDOM from "react-dom"
+import sizeMe from "react-sizeme"
+import styled, { StyledFunction } from "styled-components"
+import { fadeIn, fadeOut } from "../Assets/Animations"
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  hide?: boolean
+  maxWidth?: string
+  minWidth?: string
+  anchorRef?: HTMLElement
+  dismissOnClick?: boolean
+  message?: ((m: any) => any) | string
+  // dismissOnTimeout?: boolean
+  // other offsets? I think this is a bad idea
+}
+
+interface PositionProps {
+  top: number
+  left: number
+  width: number
+}
+
+const Positionable: StyledFunction<
+  { anchorPosition: PositionProps; ownWidth?: number; size?: any } & Props &
+    React.HTMLProps<HTMLDivElement>
+> =
+  styled.div
+
+const sizeMeOptions = {
+  monitorWidth: true,
+}
+
+const Container = Positionable`
+  position: fixed;
+  top: ${({anchorPosition}) => anchorPosition && anchorPosition.top}px;
+  left: ${({anchorPosition}) => anchorPosition && anchorPosition.left}px;
+  ${({ minWidth }) => minWidth && `min-width: ${minWidth};`}
+  ${({ maxWidth }) => maxWidth && `max-width: ${maxWidth};`}
+`
+
+const Tip = Positionable`
+  position: absolute;
+  display: ${props => (props.hide ? "none" : "block")};
+  visibility: ${props => (props.hide ? "hidden" : "visible")};
+  animation: ${props => (props.hide ? fadeOut : fadeIn)} 1s linear;
+  z-index: 1000;
+  `
+  // align-items: center;
+  // flex-direction: column;
+
+const Arrow = Positionable`
+  width: 0;
+  height: 0;
+  border-bottom: 10px solid #000;
+  border-left: 10px solid transparent;
+  border-right: 10px solid transparent;
+  position: absolute;
+  left: ${({anchorPosition}) =>
+    anchorPosition && anchorPosition.width / 2 - 10}px;
+`
+// or:
+// width: 12px;
+// height: 12px;
+// transform: rotate(-45deg);
+// background: #000;
+
+const SizelessMessage = Positionable`
+  ${props => (props.size && console.log(props.size)) || ""}
+  position: absolute;
+  top: 10px;
+  left: ${({ anchorPosition, size }) =>
+    anchorPosition && anchorPosition.width / 2 - size.width / 2}px;
+  padding: 13px 15px;
+  background: #000;
+  color: #fff;
+  font-family: "Courier New", Courier, monospace;
+  cursor: default;
+  & span {
+    cursor: pointer;
+  }
+`
```

Comment:
#minor - elsewhere in Reaction StyledComponents typically go below Class / Component definitions, at the bottom of the file